### PR TITLE
cached getVersion() and getPath()

### PR DESCRIPTION
**Pull request by Jonahss**

### CODE REVIEW

File: lib/xcode.js [hunk 1]
```diff
@@ -76,14 +79,19 @@ async function getPathFromXcodeSelect () {
 }
 
 function getPath () {
+  // cached function
+  if (cache.path) { return cache.path; }
```
Comment (jlipps) [inline lib/xcode.js:83]: yay for cached promises!

File: test/xcode-specs.js [hunk 2]
```diff
@@ -31,6 +31,18 @@ describe('xcode @skip-linux', () => {
     /\d\.\d\.*\d*/.test(version).should.be.true;
   });
 
+  it('should get the path and version again, these values are cached', async function () {
+
```
Comment (jlipps) [inline test/xcode-specs.js:35]: would love to test that retrieving from the cache takes less than 2ms or something like that

Comment (Jonahss) [inline test/xcode-specs.js:35]: ooh me too. will do.

**Comment by Jonahss** [comment]
@jlipps updated test.

**Pull request merged**